File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/Point.java
package pl.edu.agh.multiscalemodelling.engine.logic;

public class Point {

    public int x;
    public int y;
    public int z;

    public Point(int x) {

        this.x = x;
        this.y = 1;
        this.z = 1;

    }

    public Point(int x, int y) {

        this.x = x;
        this.y = y;
        this.z = 1;

    }

    public Point(int x, int y, int z) {

        this.x = x;
        this.y = y;
        this.z = z;

    }

}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/Point.scala
package pl.edu.agh.multiscalemodelling.engine.logic

class Point {
  var x = 0
  var y = 0
  var z = 0

  def this(x: Int) {
    this()
    this.x = x
    this.y = 1
    this.z = 1
  }

  def this(x: Int, y: Int) {
    this()
    this.x = x
    this.y = y
    this.z = 1
  }

  def this(x: Int, y: Int, z: Int) {
    this()
    this.x = x
    this.y = y
    this.z = z
  }
}